Update and improve introduction in zha.markdown

Hedda opened the pull request:
## Proposed change

Updated and improved the ZHA integration introduction section (including adding a couple of links to other sections in this ZHA documentation that IMHO isalso very important for beginners to read and be aware of).

Since the ZHA integration userbase/usage increased (possibly due to SkyConnect/Yellow + other factors) we see more and more new users of the ZHA integration now posting in Home Assistant's community forum every day asking for help soon after they run into common setup or environment issues when getting started with the ZHA integration and Zigbee. 

I think most of those issues occur because many of those new users did not understand the basic Zigbee fundamentals and limitations that for for better or worse users should really need to know before the begin creating a new Zigbee network and start simply adding any Zigbee devices located anywhere in their home to the ZHA integration.

I believe this improved introduction could help many new users avoid most of those basic issues or common pitfalls for Zigbee, and hopefully, they will get a better out-of-the-box user experience from the very beginning and not early in their Zigbee adoption have to go into delving deeper into Zigbee troubleshooting to correct simple setup mistakes that are caused by common Zigbee limitations.

For example; new users are not aware that they must add many Zigbee Router devices to get any kind of decent range and coverage, epecially before they start adding Zigbee End Devices.

## Type of change
<!--
    What types of changes does your PR introduce to our documentation/website?
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR.
-->

- [x] Spelling, grammar or other readability improvements (`current` branch).
- [x] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information
<!--
    Details are important, and help maintainers processing your PR.
    Please be sure to fill out additional details, if applicable.
-->

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist
<!--
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR. If you're unsure about any of them, don't hesitate to ask.
    We're here to help! This is simply a reminder of what we are going to look
    for before merging your code.
-->

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

- **Documentation**
  - Enhanced the description of the ZHA integration, emphasizing it as a hardware-independent Zigbee gateway.
  - Provided detailed guidance on connecting a Zigbee Coordinator, creating a Zigbee network, and pairing devices.
  - Highlighted compatibility and interoperability with various Zigbee devices.
  - Stressed the importance of Zigbee Router devices for network expansion and optimization.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Hedda committed:
Improve introduction in zha.markdown

Improve ZHA introduction in zha.markdown

Hedda committed:
Update zha.markdown fix lint

Incorrect usage of the terminology term: “backwards compatible”, use “backward compatible” instead.

puddly commented on source/_integrations/zha.markdown:
```diff
@@ -69,15 +69,17 @@ There is also support for Zigbee grouping of lights, switches, and fans (i.e. su
-Once ZHA has been set up and the Zigbee Coordinator radio adapter or module is configured, you will be able to directly join/pair any Zigbee device to the Zigbee network; regardless of the manufacturer and brand of that Zigbee-based product. Note, that while it is generally recommended to buy Zigbee 3.0 compliant devices as those should, in theory, offer greater interoperability, be sure of the s ...
+Because a Zigbee relies on "mesh networking" technology it depends on its Zigbee Router devices for expanding the network coverage and size. These are always mains-powered devices that will act as Zigbee routers (often wrongly referred to as signal repeaters) within the Zigbee network mesh to extend its range and expand the total amount of devices that can be added. Zigbee End Device devices on t ...
+
+You will also keep in mind that a Zigbee network needs to have many Zigbee Router devices or else your network mesh coverage will be severely limited due to the short range and poor wall penetration of Zigbee radio signals. In addition, be aware that all Zigbee devices, and especially the Zigbee Coordinator, can be very sensitive to electromagnetic interference (EMI), As such it is highly recomme ...
```
I think this last paragraph is a little redundant, since we already have a section on RF interference.

Hedda replied:
Think that at least the link should be keept as IMHO extremly important (probably solves more than half of all issues posted in Home Assistant community forum)

Hedda replied:
Changed in https://github.com/home-assistant/home-assistant.io/pull/28869/commits/e9e308f0df0a4dc4eb1b27e706b27d7bfa7b7b1d

puddly commented on source/_integrations/zha.markdown:
```diff
@@ -69,15 +69,17 @@ There is also support for Zigbee grouping of lights, switches, and fans (i.e. su
+Once ZHA has been set up and configured with a Zigbee Coordinator radio adapter or module you will be able to directly join/pair any Zigbee Router devices and Zigbee End Devices to the Zigbee network; [regardless of the manufacturer and brand of that Zigbee-based product](#knowing-which-devices-are-supported), (with few [limitations](#limitations)). While backward compatible with devices that use ...
 
-Once ZHA has been set up and the Zigbee Coordinator radio adapter or module is configured, you will be able to directly join/pair any Zigbee device to the Zigbee network; regardless of the manufacturer and brand of that Zigbee-based product. Note, that while it is generally recommended to buy Zigbee 3.0 compliant devices as those should, in theory, offer greater interoperability, be sure of the s ...
+Because a Zigbee relies on "mesh networking" technology it depends on its Zigbee Router devices for expanding the network coverage and size. These are always mains-powered devices that will act as Zigbee routers (often wrongly referred to as signal repeaters) within the Zigbee network mesh to extend its range and expand the total amount of devices that can be added. Zigbee End Device devices on t ...
```
There are dedicated signal repeater devices for sale. I don't think the comment is necessary.

```suggestion
Because a Zigbee relies on "mesh networking" technology it depends on its Zigbee Router devices for expanding the network coverage and size. These are always mains-powered devices that will act as Zigbee routers within the Zigbee network mesh to extend its range and expand the total amount of devices that can be added. Zigbee End Device devices on the other hand will not expand the range of the network as they do not pass along any signals/messages from other devices, these are usually all battery-operated devices, but not always. You should therefore make sure that your Zigbee network has enough Zigbee Router devices, and always begin by adding those before adding Zigbee End devices when getting started.
```

Hedda replied:
Changed in https://github.com/home-assistant/home-assistant.io/pull/28869/commits/e9e308f0df0a4dc4eb1b27e706b27d7bfa7b7b1d

puddly commented on source/_integrations/zha.markdown:
```diff
@@ -69,15 +69,17 @@ There is also support for Zigbee grouping of lights, switches, and fans (i.e. su
+In the case of this ZHA integration, the Zigbee Coordinator is a physical radio adapter or hardware module, that contains a microcontroller that runs the Zigbee stack which the ZHA integration uses to manage and communicate with the Zigbee network and all its devices. The Zigbee Coordinator will act as the central device in your Zigbee network. Its main responsibility from an end-user's point-of- ...
 
-Before installing the ZHA integration, you will need to buy and connect a Zigbee Coordinator radio adapter or hardware module, those usually come in the form of a USB dongle that plugs directly into the computer that is running the Home Assistant installation. The ZHA integration can work with many different "Zigbee Coordinator" adapters, however, be sure to read the respective sections below abo ...
+Once ZHA has been set up and configured with a Zigbee Coordinator radio adapter or module you will be able to directly join/pair any Zigbee Router devices and Zigbee End Devices to the Zigbee network; [regardless of the manufacturer and brand of that Zigbee-based product](#knowing-which-devices-are-supported), (with few [limitations](#limitations)). While backward compatible with devices that use ...
```
Terminology like "has been set up and configured" and "Zigbee Coordinator radio adapter or module" is unnecessarily verbose/redundant and hurts readability. Let's stick to simpler and shorter language, there's no need to be 100% precise in an introduction.

One possibility (roughly):

```suggestion
Once ZHA has been set up with a Zigbee radio, you will be able to join Zigbee devices to the network. While most devices will join, not all functionality [will always be supported out of the box](#knowing-which-devices-are-supported). It is generally recommended to buy Zigbee 3.0 devices. Zigbee 3.0 certified devices usually offer better interoperability, however devices that use manufacturer-specific extensions to add non-standard functions and features will still need [device-specific code to fully work with ZHA](#how-to-add-support-for-new-and-unsupported-devices).
```

Hedda replied:
Updated to at least remove most jargon.

Hedda replied:
Changed in https://github.com/home-assistant/home-assistant.io/pull/28869/commits/e9e308f0df0a4dc4eb1b27e706b27d7bfa7b7b1d

puddly reviewed (changes requested):
Thanks!

Could you make the changes a bit more compact? There are many phrases that incorporate too much jargon and can often times be entirely removed without changing any meaning:

 - "Zigbee Coordinator radio adapter or hardware module"
 - "physical radio adapter"
 - "from an end-user's point-of-view"

Specification revisions and acronyms are a little too techincal for an introduction so let's keep it as simple as possible.

Hedda committed:
Update zha.markdown

Rewrote to remove some jargon

Hedda commented:
> Could you make the changes a bit more compact? There are many phrases that incorporate too much jargon and can often times be entirely removed without changing any meaning:
> 
> * "Zigbee Coordinator radio adapter or hardware module"
> * "physical radio adapter"
> * "from an end-user's point-of-view"
> 
> Specification revisions and acronyms are a little too techincal for an introduction so let's keep it as simple as possible.

Updated to at least remove most technical jargon + acronyms, and split only more sections to try to get it less compact

Hedda committed:
Update zha.markdown

Hedda committed:
Merge branch 'current' into patch-93

Hedda commented:
@puddly I have tried to make all requested changes so wondering if there is any chance to get this reviewed again?

Hedda committed:
Merge branch 'current' into patch-93

Hedda committed:
Merge branch 'current' into patch-93

puddly commented on source/_integrations/zha.markdown:
```diff
@@ -73,15 +73,21 @@ In addition, it has support for "Zigbee groups" that enable native on-device gro
+Once ZHA has been set up with a Zigbee Coordinator it will automatically create a Zigbee network and you will be able to join/pair any Zigbee devices. With only a few [limitations](#limitations), most devices will join/pair directly regardless of brand and manufacturer. It is generally recommended to buy Zigbee 3.0 compliant devices if possible, though a new Zigbee Coordinator is also backward co ...
 
-Once ZHA has been set up and the Zigbee Coordinator radio adapter or module is configured, you will be able to directly join/pair any Zigbee device to the Zigbee network; regardless of the manufacturer and brand of that Zigbee-based product. Note, that while it is generally recommended to buy Zigbee 3.0 compliant devices as those should, in theory, offer greater interoperability, be sure of the s ...
+The Zigbee Coordinator will act as the central device in your Zigbee network. In the case of this ZHA integration, the Zigbee Coordinator has a microcontroller and runs the Zigbee stack which is used to manage the Zigbee network and communicate with all its devices. Its main responsibility is the joining/pairing of devices to the network and then controlling them. Being the physical network bridg ...
```
I think this paragraph can be removed entirely.

Hedda replied:
OK removed

puddly commented on source/_integrations/zha.markdown:
```diff
@@ -73,15 +73,21 @@ In addition, it has support for "Zigbee groups" that enable native on-device gro
+
+Zigbee End Device devices will not expand the range of the network as they do not route any messages from other devices. These are usually all battery-operated devices, but not always as some Zigbee End Devices can sometimes be mains-powered devices. Always begin adding Zigbee Router devices before Zigbee End Devices when creating a new Zigbee network. It is highly recommended that you read and f ...
+
+Optionally you can configure Zigbee grouping of lights, switches, and fans (i.e. support for commanding multiple devices as a single group). In addition, you also have the [option to bind some Zigbee devices](#binding-and-unbinding), like a remote, directly to a device or group (i.e. define relations between two or more Zigbee devices that will work even when the Zigbee Coordinator and ZHA integr ...
```
This paragraph too.

Hedda replied:
Removed, rewritten and restructuring based on feedback

puddly reviewed (changes requested):
Keep in mind that this documentation is for the ZHA **integration**, not for Zigbee in general!

Let's focus on *paring down* the documentation. Right now, it's so long and dense that a typical user of Home Assistant can't just open it up and look at it without being overwhelmed. We don't need to cover every use case and every possible question someone can possibly have about Zigbee in the documentation.

Right now, we don't have a place for in-depth Zigbee documentation like this. Perhaps you can instead extend the Zigpy wiki and we then migrate the content in the future to a generated documentation website?

Hedda committed:
Update zha.markdown

Update introduction based on feedback.

Hedda commented:
> Keep in mind that this documentation is for the ZHA **integration**, not for Zigbee in general!
> 
> Let's focus on _paring down_ the documentation.

OK, I now tried restructuring and rewritten the introduction section based on that feedback.

Hedda committed:
Merge branch 'current' into patch-93

Hedda committed:
Update zha.markdown

Hedda committed:
Merge branch 'current' into patch-93

TheJulianJES commented on source/_integrations/zha.markdown:
```diff
@@ -73,15 +73,15 @@ In addition, it has support for "Zigbee groups" that enable native on-device gro
 ## Introduction
 
-ZHA {% term integration %} is a Zigbee gateway implementation that follows the standard Zigbee 3.0 specification (and earlier revisions). In Zigbee, there are three different device types: Zigbee Coordinator (ZC), Zigbee Router (ZR), and Zigbee End Device (ZED). A Zigbee network always has one (and no more) Zigbee Coordinator, however, a Zigbee network can have multiple Zigbee Routers and multipl ...
+This ZHA integration is a hardware independent Zigbee gateway implementation that can replace most proprietary Zigbee gateways/bridges/hubs/controllers. It supports pairing/joining devices that use the Zigbee 3.0 standard and earlier Zigbee standards for home automation and lighting. It follows the Zigbee PRO specification, a low-power wireless communication protocol using small digital radios to ...
```
I don't think should mention "Zigbee 3.0", "earlier Zigbee standards", and "Zigbee PRO" directly in the introduction. We probably don't need to at all.

TheJulianJES commented on source/_integrations/zha.markdown:
```diff
@@ -73,15 +73,15 @@ In addition, it has support for "Zigbee groups" that enable native on-device gro
+Before installing the ZHA integration in Home Assistant, you need to connect a Zigbee Coordinator radio adapter, that will act as the physical network bridge interface and single point of security controller for your Zigbee network. Those normally come in the form of a USB dongle that plugs directly into the same computer that is running your Home Assistant installation. The ZHA integration is co ...
 
-In the case of ZHA, the Zigbee Coordinator is a radio adapter or hardware module, that contains a microcontroller that runs the Zigbee protocol stack which the ZHA {% term integration %} uses to manage and communicate with a Zigbee network and its devices.
+Once ZHA has been set up with a Zigbee Coordinator it will automatically create a Zigbee network and you will be able to join/pair any Zigbee Router devices and Zigbee End Devices. With only a few [limitations](#limitations), most devices will join/pair directly regardless of brand and manufacturer. It is generally recommended to buy Zigbee 3.0 compliant devices if possible because those offer be ...
```
Again, not sure we need to recommend buying "Zigbee 3.0" devices. It basically means nothing in terms of compatibility now (unfortunately) and most Tuya products are advertised as "Zigbee 3.0", even though many aren't compatible.

TheJulianJES commented on source/_integrations/zha.markdown:
```diff
@@ -73,15 +73,15 @@ In addition, it has support for "Zigbee groups" that enable native on-device gro
+Because a Zigbee relies on "mesh networking" technology it depends heavily on having [Zigbee Router devices](#using-router-devices-to-add-more-devices) to expand the network coverage and extend its size. These are always mains-powered devices that route messages to other devices that are located close to them within the Zigbee network mesh to improve the range and increase the total amount of dev ...
 
-Once ZHA has been set up and the Zigbee Coordinator radio adapter or module is configured, you will be able to directly join/pair any Zigbee device to the Zigbee network; regardless of the manufacturer and brand of that Zigbee-based product. Note, that while it is generally recommended to buy Zigbee 3.0 compliant devices as those should, in theory, offer greater interoperability, be sure of the s ...
+Optionally you can configure Zigbee grouping of lights, switches, and fans (i.e. support for commanding multiple devices as a single group). In addition, you also have the [option to bind some Zigbee devices](#binding-and-unbinding), like binding a button on a remote control directly to a device or group (i.e. define relations between two or more Zigbee devices that will work even when the Zigbee ...
```
I'm not sure we need to mention groups and binding in the introduction.

TheJulianJES commented on source/_integrations/zha.markdown:
```diff
@@ -73,15 +73,15 @@ In addition, it has support for "Zigbee groups" that enable native on-device gro
+Once ZHA has been set up with a Zigbee Coordinator it will automatically create a Zigbee network and you will be able to join/pair any Zigbee Router devices and Zigbee End Devices. With only a few [limitations](#limitations), most devices will join/pair directly regardless of brand and manufacturer. It is generally recommended to buy Zigbee 3.0 compliant devices if possible because those offer be ...
 
-Before installing the ZHA {% term integration %}, you will need to buy and connect a Zigbee Coordinator radio adapter or hardware module, those usually come in the form of a USB dongle that plugs directly into the computer that is running the Home Assistant installation. The ZHA {% term integration %} can work with many different "Zigbee Coordinator" adapters, however, be sure to read the respect ...
+Because a Zigbee relies on "mesh networking" technology it depends heavily on having [Zigbee Router devices](#using-router-devices-to-add-more-devices) to expand the network coverage and extend its size. These are always mains-powered devices that route messages to other devices that are located close to them within the Zigbee network mesh to improve the range and increase the total amount of dev ...
```
> a Zigbee

Hedda committed:
Merge branch 'current' into patch-93

Hedda committed:
Update zha.markdown based on feedback from TheJulianJES

Update zha.markdown based on feedback from TheJulianJES

Hedda commented:
@TheJulianJES  OK the latest commit made changes based on your feedback so hope that the introduction reads better now.

Hedda committed:
Merge branch 'current' into patch-93

puddly commented on source/_integrations/zha.markdown:
```diff
@@ -73,15 +73,13 @@ In addition, it has support for "Zigbee groups" that enable native on-device gro
+This ZHA integration is a hardware-independent Zigbee gateway implementation that can replace most proprietary Zigbee gateways/bridges/hubs/controllers. Zigbee is a low-bandwidth communication protocol that relies on using small low-power digital radios to connect compatible devices to local Zigbee wireless private area networks. ZHA will create a single Zigbee network to which you can then pair/ ...
 
-A Zigbee Coordinator is the central device in a Zigbee network that manages and controls the network. It acts as a physical interface for the Zigbee wireless protocol, as well as being responsible for maintaining the Zigbee network topology and ensuring secure and efficient communication between Zigbee devices. Zigbee Router devices are mains-powered and will act as Zigbee signal repeaters within ...
+Before installing the ZHA integration in Home Assistant, you need to connect a Zigbee Coordinator radio adapter, that will act as the physical network bridge interface and single point of security controller for your Zigbee network. Those normally come in the form of a USB dongle that plugs directly into the same computer that is running your Home Assistant installation. The ZHA integration is co ...
```
```suggestion
Before installing the ZHA integration in Home Assistant, you need to connect a Zigbee Coordinator radio adapter that will connect to your Zigbee network. Those normally come in the form of a USB dongle that plugs directly into the same computer that is running your Home Assistant installation. The ZHA integration is compatible with many different "Zigbee Coordinator" adapters from various manufacturers. Be sure to [note the recommendations in the respective sections below before buying a Zigbee Coordinator](#compatible-hardware). A Zigbee network always needs to have one Zigbee Coordinator (it can never have more than one), and Zigbee devices can never be connected to more than a single Zigbee network, however, a Zigbee network can have multiple "Zigbee Router" devices and "Zigbee End Device" products.
```

puddly reviewed (changes requested):
This PR looks good, thank you.

Just one small wording change and it's good to merge.

puddly committed:
Update source/_integrations/zha.markdown

Hedda commented on source/_integrations/zha.markdown:
```diff
@@ -73,15 +73,13 @@ In addition, it has support for "Zigbee groups" that enable native on-device gro
-Before installing the ZHA {% term integration %}, you will need to buy and connect a Zigbee Coordinator radio adapter or hardware module, those usually come in the form of a USB dongle that plugs directly into the computer that is running the Home Assistant installation. The ZHA {% term integration %} can work with many different "Zigbee Coordinator" adapters, however, be sure to read the respect ...
-
-Once ZHA has been set up and the Zigbee Coordinator radio adapter or module is configured, you will be able to directly join/pair any Zigbee device to the Zigbee network; regardless of the manufacturer and brand of that Zigbee-based product. Note, that while it is generally recommended to buy Zigbee 3.0 compliant devices as those should, in theory, offer greater interoperability, be sure of the s ...
+Note that because Zigbee relies on "mesh networking" technology it depends heavily on having [Zigbee Router devices](#using-router-devices-to-add-more-devices) to expand the network coverage and extend its size. These are always mains-powered devices that route messages to other devices that are located close to them within the Zigbee network mesh to improve the range and increase the total amoun ...
```
I am not going to add a visual diagram to this PR.

puddly replied:
You can ignore the bot.

Hedda commented on source/_integrations/zha.markdown:
```diff
@@ -73,15 +73,13 @@ In addition, it has support for "Zigbee groups" that enable native on-device gro
+Before installing the ZHA integration in Home Assistant, you need to connect a Zigbee Coordinator radio adapter that will connect to your Zigbee network. Those normally come in the form of a USB dongle that plugs directly into the same computer that is running your Home Assistant installation. The ZHA integration is compatible with many different "Zigbee Coordinator" adapters from various manufac ...
 
-In the case of ZHA, the Zigbee Coordinator is a radio adapter or hardware module, that contains a microcontroller that runs the Zigbee protocol stack which the ZHA {% term integration %} uses to manage and communicate with a Zigbee network and its devices.
+Once ZHA has been set up with a Zigbee Coordinator it will automatically create a Zigbee network and you will be able to join/pair any Zigbee Router devices and Zigbee End Devices. With only a few [limitations](#limitations), most devices will join/pair directly regardless of brand and manufacturer. Technically almost all devices that are compliant with the official Zigbee specifications should o ...
```
Not relevant.

Hedda commented on source/_integrations/zha.markdown:
```diff
@@ -73,15 +73,13 @@ In addition, it has support for "Zigbee groups" that enable native on-device gro
+This ZHA integration is a hardware-independent Zigbee gateway implementation that can replace most proprietary Zigbee gateways/bridges/hubs/controllers. Zigbee is a low-bandwidth communication protocol that relies on using small low-power digital radios to connect compatible devices to local Zigbee wireless private area networks. ZHA will create a single Zigbee network to which you can then pair/ ...
 
-A Zigbee Coordinator is the central device in a Zigbee network that manages and controls the network. It acts as a physical interface for the Zigbee wireless protocol, as well as being responsible for maintaining the Zigbee network topology and ensuring secure and efficient communication between Zigbee devices. Zigbee Router devices are mains-powered and will act as Zigbee signal repeaters within ...
+Before installing the ZHA integration in Home Assistant, you need to connect a Zigbee Coordinator radio adapter that will connect to your Zigbee network. Those normally come in the form of a USB dongle that plugs directly into the same computer that is running your Home Assistant installation. The ZHA integration is compatible with many different "Zigbee Coordinator" adapters from various manufac ...
```
Had such an explanation before in this PR but where asked to remove it so I did.

Hedda commented on source/_integrations/zha.markdown:
```diff
@@ -73,15 +73,13 @@ In addition, it has support for "Zigbee groups" that enable native on-device gro
 ## Introduction
 
-ZHA {% term integration %} is a Zigbee gateway implementation that follows the standard Zigbee 3.0 specification (and earlier revisions). In Zigbee, there are three different device types: Zigbee Coordinator (ZC), Zigbee Router (ZR), and Zigbee End Device (ZED). A Zigbee network always has one (and no more) Zigbee Coordinator, however, a Zigbee network can have multiple Zigbee Routers and multipl ...
+This ZHA integration is a hardware-independent Zigbee gateway implementation that can replace most proprietary Zigbee gateways/bridges/hubs/controllers. Zigbee is a low-bandwidth communication protocol that relies on using small low-power digital radios to connect compatible devices to local Zigbee wireless private area networks. ZHA will create a single Zigbee network to which you can then pair/ ...
```
Not needed in the introduction I think as should be in the configuration section instead.
